devlib/target.py

```diff
@@ -153,6 +155,7 @@ def __init__(self,
                  ):
         self.connection_settings = connection_settings or {}
         self.platform = platform or Platform()
+        self.connection_settings['platform'] = self.platform
```

Comment:
Correct. However, they're are already expecting the other arguments. this is a new argument being added. So it needs to be added to all connection implementations.